flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/MySqlValidator.java

```diff
@@ -157,4 +161,36 @@ private void checkBinlogRowImage(JdbcConnection connection) throws SQLException
                             rowImage, BINLOG_FORMAT_IMAGE_FULL));
         }
     }
+
+    /** Check whether the server timezone matches the configured timezone. */
+    private void checkTimeZone(JdbcConnection connection) throws SQLException {
+        String timeZoneProperty = dbzProperties.getProperty("database.serverTimezone");
+        if (timeZoneProperty == null) {
+            LOG.warn(
+                    "{} is not set, which might cause data inconsistencies for time-related fields.",
+                    SERVER_TIME_ZONE.key());
+            return;
+        }
+
+        ZoneId timeZone = ZoneId.of(timeZoneProperty);
+        int timeZoneOffsetInSeconds =
+                timeZone.getRules().getOffset(LocalDateTime.now()).getTotalSeconds();
```

Comment:
I have some concerns about the validation:
(1) The timezone may have multiple validate offsets.
(2)`timeDiffInSeconds != timeZoneOffsetInSeconds` it too strictly because the returned value of `TIME_TO_SEC(TIMEDIFF(NOW(), UTC_TIMESTAMP))` may be different seconds in some DBMS.

Reply:
Thanks for the insights.

Yes, for question 1), take Los Angeles for example, when Daylight Saving Time begins, the actual timezone shifts from 'Pacific Standard Time (UTC-8)' to 'Pacific Daylight Time (UTC-7)', and some DBAs choose to change timezone accordingly, others might choose to stay with the standard timezone all year long. Thus we need to allow both cases, and I have loosened the check in the latest commit.

For the question 2), since this MR only focuses on MySQL, I have tested on 5.6, 5.7, 8.0, it seems that the returned value of `TIME_TO_SEC(TIMEDIFF(NOW(), UTC_TIMESTAMP))` behaves as expected so far. I wonder if there are other cases that I haven't found out?

Reply:
For the question 2  I mean other MySQL vendors like PolarDB-x/TDSQL and so on. But I think we can focus on MySQL currently and adjust our strategy in the fututre if other vendors can not pass the check.

Reply:
Hi @leonardBang , you are right that the newly added PolarDB-X test case throw exceptions like '14c6666538c00000][172.17.0.2:8527][polardbx]ERR-CODE: [PXC-4518][ERR_VALIDATE] : Column 'UTC_TIMESTAMP' not found in any table' when executing TIMEDIFF queries, because it mistakenly treats `UTC_TIMESATMP` as a column name.

After changing `UTC_TIMESTAMP` to `UTC_TIMESTAMP()`, this problem is settled. Anyway, TDSQL-C behaves the same as MySQL and treats `UTC_TIMESTAMP` the same as `UTC_TIMESTAMP()` call.